# Precision Medicine Knowledge Graph (PrimeKG)

**A multimodal knowledge graph for precision medicine analyses**
## What?

Built with 20 high-quality data resources (+ some additional of biological and clinical information) to curate this knowledge graph. Across 129,375 nodes and 4,050,249 relationships, PrimeKG captures information on ten major biological scales.

The data resources provide widespread coverage of bio- medical entities, including proteins, genes, drugs, diseases, anatomy, biological processes, cellular components, molecular functions, exposures, disease phenotypes and drug side effects

![Pasted image 20250930151538](Pasted%20image%2020250930151538.png)
![Pasted image 20250930145849](Pasted%20image%2020250930145849.png)
![Pasted image 20250930184129](Pasted%20image%2020250930184129.png)
## Why?

Biomedical knowledge about complex dis- eases comes from different organizational scales, including genomics, transcriptomics, proteomics, molecular functions, intra- and inter-cellular pathways, phenotypes, therapeutics, and environmental effects. For any given disease, information from these organizational scales is scattered across publications, non-standardized data repositories, evolving ontologies, and clinical guidelines. Developing networked relationships between these sources can support research in precision medicine.

While many primary data resources contain information about diseases, consolidating them into a comprehensive, disease-rich, and functional knowledge graph presents three challenges. 
- First, existing approaches to network analysis of diseases require expert review and curation of data in the knowledge graph
- Second, there lacks a consistent representation of diseases across biomedical datasets and clinical guidelines.
- Third, the definition of diseases as discrete and distinct units of analysis remains medically and scientifically ambiguous (e.g. ASD)

## Datasets

- [Bgee] - Gene expression across multiple animal species
- [Comparative Toxicogenomics Database] - Environmental exposures on human health
- [DisGeNET] - Relationships between genes and human disease
- [Disease Ontology] - Groups diseases in meaningful clusters using clinically relevant characteristics. Allows the exploration of disease nodes in a medically meaningful format.
- [Drugbank] - Pharmaceutical knowledge, synergistic drug interactions
- [Drug Central] - Drug-disease interactions
- [Entrez Gene] - Gene specific information
- [Gene Ontology] - Describes molecular functions, cellular components, and biological processes
- [Human Phenotype Ontology] - Phenotypic abnormalities found in diseases
- [Mayo Clinic] - Symptoms, causes, risk factors, complications, and prevention of 2,227 diseases and conditions
- [MONDO disease ontology] - Defines diseases; harmonizes data from a wide range of ontologies
- [Orphanet] - Knowledge of rare diseases
- [Four Sources of Physical Protein-Protein Interaction] - Verifiable interaction between proteins
- [Reactome Pathway Database] - Curated database for pathways
- [Side Effect Knowledgebase] - Adverse drug reactions
- [Uberon multi-species anatomy ontology] - Human anatomy
- [UMLS Knowledgebase] - Biomedical and health context